### DISCUSSION TITLE
[Example] Color based on multiple states

### DISCUSSION THREAD

**Discussion by MrBearPresident** [Clooos/Bubble-Card#929] [Share your custom styles, templates and dashboards]
![Screen Recording 2024-11-16 153049](https://github.com/user-attachments/assets/e9c840ee-cdba-44b5-90ff-a8fb07859af3)

```
type: custom:bubble-card
card_type: button
entity: person.me
button_type: state
show_state: true
styles: |-
  .bubble-button-background {
    background-color: ${
      {
        'home': '#8AD926FF',
        'away': '#1982C4FF',
        'Summer House': '#CDCA3AFF',
        'Parents House': '#6A4C93AA',
      }[state] ?? '#ff0000AA'
    };
  }
card_layout: large
```

**Note:** The color code after the `??` is the default value, when the state is not found in the list. 

### Disclamer: I did not make this, but revised what @j_d posted.
- https://community.home-assistant.io/t/bubble-card-a-minimalist-card-collection-for-home-assistant-with-a-nice-pop-up-touch/609678/1777?u=mrbearpresident

**Comment by johndoe0815** [comment]
Would something like this also be possible with other elements than color, e.g. icon, and also based on different states than the one of the entity?

Currently I have for example the following custom style code to put a logo into a sub-button based on the media source currently plaing:

```
${subButtonIcon[0].setAttribute("icon", hass.states['media_player.fire_tv_cube'].attributes.source == 'Disney+' ? 'phu:disney-plus' 
: hass.states['media_player.fire_tv_cube'].attributes.source == 'Netflix' ? 'phu:netflix' 
: hass.states['media_player.fire_tv_cube'].attributes.source == 'Prime Video (FireTV)' ? 'phu:prime-video' 
: hass.states['media_player.fire_tv_cube'].attributes.source == 'Kodi' ? 'mdi:kodi' 
: '')}
```

**Reply by MrBearPresident** [reply]
Yes it is possible, by the same basic technique but not in the same way. 

```
  #Place all the DOM changing stuff on the bottom

  ${subButtonIcon[0].setAttribute("icon",
      {
      'Disney': 'phu:disney-plus',
      'Netflix': 'phu:netflix',
      'Prime Video (FireTV)': 'phu:prime-video',
      'Kodi': 'mdi:kodi',
    }[hass.states['media_player.fire_tv_cube'].attributes.source] ?? 'mdi:television'
  )}
```

**Reply by johndoe0815** [reply]
Awesome, thanks!
Works perfectly and is much easier to maintain. This is really not intuitive, so good to have such code snippets 👍 
![grafik](https://github.com/user-attachments/assets/5aa767ef-3cf4-4a90-b1c4-6ba882d0cd7c)

**Comment by MilesTEG1** [comment]
Hello,
This is working fine for the main button.
But for sub-button with different entity, this is not working, because the `state` is the one from the main button.
How can I achieve to use the entity state from the sub-button?
Thanks in advance 😊

**Reply by MrBearPresident** [reply]
https://github.com/Clooos/Bubble-Card/discussions/929#discussioncomment-11511998

**Reply by MilesTEG1** [reply]
> https://github.com/Clooos/Bubble-Card/discussions/929#discussioncomment-11511998

Thanks.
I missed it, sorry.

For what I want to do, change text color and background-color, do you think this will work?

```yaml
${subButton[0].setAttribute("color",
      {
      'Blanc': 'rgba(255, 255, 255, 1)',
      'Bleu': 'rgba(0, 0, 255, 1)',
      'Rouge': 'rgba(255, 0, 0, 1)',
    }[hass.states['sensor.rte_tempo_couleur_actuelle'].attributes.source] ?? 'rgba(126, 126, 126, 1)'
  )}

${subButton[0].setAttribute("Background-Color",
      {
      'Blanc': 'rgba(255, 255, 255, 1)',
      'Bleu': 'rgba(0, 0, 255, 1)',
      'Rouge': 'rgba(255, 0, 0, 1)',
    }[hass.states['sensor.rte_tempo_couleur_actuelle'].attributes.source] ?? 'rgba(126, 126, 126, 1)'
  )}

```

**Reply by MrBearPresident** [reply]
No you want to use the state not an attribute. 

so it should be:
```
${subButton[0].setAttribute("color",
      {
      'Blanc': 'rgba(255, 255, 255, 1)',
      'Bleu': 'rgba(0, 0, 255, 1)',
      'Rouge': 'rgba(255, 0, 0, 1)',
    }[hass.states['sensor.rte_tempo_couleur_actuelle'].state] ?? 'rgba(126, 126, 126, 1)'
  )}

${subButton[0].setAttribute("Background-Color",
      {
      'Blanc': 'rgba(255, 255, 255, 1)',
      'Bleu': 'rgba(0, 0, 255, 1)',
      'Rouge': 'rgba(255, 0, 0, 1)',
    }[hass.states['sensor.rte_tempo_couleur_actuelle'].state] ?? 'rgba(126, 126, 126, 1)'
  )}
```

**Reply by MilesTEG1** [reply]
Thanks!
I look into it when I have some time 😊

**Comment by vednolacni** [comment]
Is this still working?I copied the code and changed the person entity, and i have state home but the colors are not working.

**Reply by MrBearPresident** [reply]
Hey there was a small error in the code I updated it.

**Reply by vednolacni** [reply]
Its working now. Thanks!

**Comment by adnansarajlic** [comment]
This one does not seem to work with latest beta 9, anybody else seen that? Colors stay as standard.
@Clooos @MrBearPresident 

.bubble-button-background {
    background-color: ${
    {
      'LOW': '#449548',
      'NORMAL': '#1982C4FF',
      'HIGH': '#B13A33',
    }[hass.states['sensor.nordpool_elpris'].attributes.price_level] ?? '#ff0000AA'
  }

**Reply by Clooos** [reply]
I will check that, thank you for the feedback!

**Reply by damomato** [reply]
I'm having the same issue, also on beta 9. I was going to downgrade to see if that worked but decided to wait until Clooos investigates.

Just want to check if my config looks correct first of all? I haven't done this sort of styling before. Colours just taken from an above example.

.bubble-button-background {
  background-color: ${
    {
      'Cool': '#8AD926FF',
      'Heat': '#1982C4FF',
      'Dry': '#CDCA3AFF',
    }[hass.states['climate.house_aircon'].state] ?? '#ff0000AA'
}

**Reply by damomato** [reply]
Actually nevermind I reverter to v2.4 and it's working fine, so definitely something wrong with the 2.5 beta

**Comment by damomato** [comment]
Would someone be able to help me make a conditional background colour using the states of two entities? The logic would be

```
if [current entity] state is 'open' then 
  if ['climate.house_aircon'] state is
  'Cool' then 'blue'
  'Heat' then 'red'
  'Dry' then 'green'
  else 'grey'
else 'inherit'
```

The reason being that the aircon zone damper is a separate entity to the ducted house aircon, but I only want the colours to show when the relevant mode is active.

Thanks in advance

**Comment by Ramss3s** [comment]
working on something similar 
<img width="466" height="56" alt="image" src="https://github.com/user-attachments/assets/bdc77c0d-e84a-4fff-9774-37a6d8a047a3" />

**Reply by Ramss3s** [reply]
automatic piks up the zones and the icon asisgned in HS Zones ...

**Reply by Ramss3s** [reply]
@Clooos  have something similar ... and i do not know if is moral to share it.

**Reply by Clooos** [reply]
My badge's module is the main entry point to my Patreon page, so I would love to not see people duplicating my own modules on my own store 😅

So thank you so much for your understanding 🙏

**Reply by Ramss3s** [reply]
@Clooos i'm new to github any way to DM ?

**Reply by Clooos** [reply]
I've never found a way myself, but that would be great. But you can DM me on Reddit if you want. 

https://www.reddit.com/u/Clooooos
